### ISSUE TITLE
[TRACKER] Cyclic dependencies related issues

### ISSUE THREAD

**Issue by dalexeev** [godotengine/godot#80877]
### Godot version

4.2 dev

### System information

All

### Issue description

Anything related to cyclic dependencies/requirements of resources. Do not confuse with cyclic references (`RefCounted`).

```[tasklist]
### Bugs
- [ ] #64330
- [ ] #70985
- [ ] #74253
- [ ] #74837
- [ ] #76641
- [ ] #77090
- [ ] #77436
- [ ] #78128
- [ ] #79545
- [ ] #80043
- [ ] #80324
- [ ] #80379
- [ ] #80645
- [ ] #80714
```

```[tasklist]
### PRs
- [ ] #80854
```

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

**Comment by Maran23** [comment]
One more issue with cyclic dependencies:
https://github.com/godotengine/godot/issues/87397

**Comment by yoont4** [comment]
This tracker is nice to have, but is there also a place that aggregates design suggestions for avoiding this in the first place? I've got a fairly large project, and even slight changes, like referencing a type can suddenly cause a series of seemingly unrelated files to become corrupted. e.g. I have a global `PlayerState` autoload, and referencing something as simple as a resource type will suddenly make my units become corrupted with errors like:
```
E 0:00:01:0779   ResourceLoaderText::_parse_ext_resource: res://core/units/unit.tscn:502 - Parse Error: [ext_resource] referenced non-existent resource at: res://core/components/drag_and_drop/drag_component_2d.gd
  <C++ Source>   scene\resources\resource_format_text.cpp:163 @ ResourceLoaderText::_parse_ext_resource()
```
This is after referencing an `ItemBlueprint` type, which has nothing to do with my Units! I literally have no idea how to make forward progress on my projects anymore, because I have no idea when something is just going to blow up.

**Comment by dalexeev** [comment]
> This tracker is nice to have, but is there also a place that aggregates design suggestions for avoiding this in the first place?

@yoont4 No, trackers are usually not intended for general discussion, only for grouping/classifying related issues. You can write a comment if you notice missing issues, and a maintainer will add them to the first post.

For general discussion, please use the appropriate Godot Contributors Chat channel (in this case `#gdscript`) or the issue most relevant to your topic from the list, if any (in this case it is quite difficult to do since there are a lot of issues).

**Comment by vibrunazo** [comment]
Should #78074 be added here?

**Comment by vvvvvvitor** [comment]
#92326 was superseded by #93346.

**Comment by p10tr3k** [comment]
https://github.com/godotengine/godot/issues/99333
